api/client/client.go

```diff
@@ -227,6 +227,9 @@ func (c *Client) DownloadDirectory(hash, path, destDir, credentials string) erro
 
 	uri := c.Gateway + "/bzz:/" + hash + "/" + path
 	req, err := http.NewRequest("GET", uri, nil)
+	values := req.URL.Query()
+	values.Add("publisher", publisher)
```

Comment:
these two can be done in one line, no? as @zelig suggested

Reply:
not necessarily, the request does not update if it's done in one line (did manual testing and the query param is not reaching correctly).
It must be encoded again after being added to the query.

There multiple examples in the codebase that use it like this.
The add is correct, but if the encoding is missing, it will not pass the query param.

Example client.go line 553
```golang
q := req.URL.Query()
q.Set("defaultpath", defaultPath)
req.URL.RawQuery = q.Encode()
```

Reply:
taking your example, does this not work?
```golang
q := req.URL.Query().Set("defaultpath", defaultPath)
req.URL.RawQuery = q.Encode()
```

Reply:
```
Error
req.URL.Query().Add("publisher", publisher) (no value) used as value
```
It's compiling in a single line

Reply:
tried it, and it does not.

resolving